modules/installation-azure-user-infra-uploading-rhcos.adoc

```diff
@@ -79,7 +95,7 @@ that matches your {product-title} version if it is available.
 ----
 $ az storage container create --name vhd --account-name ${CLUSTER_NAME}sa --account-key ${ACCOUNT_KEY}
```

Comment:
So there is a bit of an issue with commands like this and like the one above:
```
$ export ACCOUNT_KEY=`az storage account keys list -g ${RESOURCE_GROUP} --account-name ${CLUSTER_NAME}sa --query "[0].value" -o tsv`
```
(GitHub will not let me leave a comment on the line above). 

The issue is that in the UPI workflow is that these variables `$CLUSTER_NAME`, `$RESOURCE_GROUP` are defined as part of the UPI flow, but don't exist in the IPI flow.

Also, these values are not linked to a particular install. The users just need to have uploaded an RHCOS image once, so we should try to make that clear in the docs.

Reply:
I am not sure that I am following you here. Are you suggesting that we include the step from the UPI doc to export the variables from the `install-config.yaml` file? If not, I would need some help understanding what the revised procedure would be.

Given that users "just need to upload an RHOCS image once", would all of the steps still be applicable, but would we simply omit any option from the sample commands that reference ` $CLUSTER_NAME` , `$RESOURCE_GROUP` and `AZURE_REGION`?

Reply:
>  Are you suggesting that we include the step from the UPI doc to export the variables from the install-config.yaml file?

No, because I want to avoid any confusion that the rhcos image upload depends on the install config. I think it is more accurate to say the install config depends on the RHCOS image having been uploaded. 

>  but would we simply omit any option from the sample commands that reference $CLUSTER_NAME , $RESOURCE_GROUP and AZURE_REGION?

Yes I think this sounds like a good solution and we are on the same page. To put it another way: In the context of IPI `$CLUSTER_NAME`, `$RESOURCE_GROUP`, & `$AZURE_REGION`  used in these commands are undefined. I think we want to avoid confusion so that customers think this RHCOS image needs to go in their openshift cluster resource group, for example, which isn't true.

In general the steps may be a bit too precise or, even, prescriptive. For instance, we tell the users to create a storage account. They don't necessarily need to... they could use an existing storage account. It might be better to a bit more vague. For example, here are [similar steps for OpenStack](https://docs.openshift.com/container-platform/4.9/installing/installing_openstack/installing-openstack-installer-restricted.html#installation-creating-image-restricted_installing-openstack-installer-restricted).

I would suggest reorganizing the steps like this:
1. Download (combine current steps 3 & 5). There is an error in the command in step 3. It should be: `export COMPRESSED_VHD_URL=$(openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.artifacts.azurestack.formats."vhd.gz".disk.location')`
2. Decompress (current step 6)
3. Upload (rephrase current step 7 as an example): 
```
az storage blob upload --account-name <rhcos-storage-account-name> --account-key <rhcos-storage-account-key> -c <rhcos-storage-account-container> -n "rhcos.vhd" -f rhcos-<rhcos_version>-azurestack.x86_64.vhd
```

I think it would be fine to even exclude the example command from the last step. Heck, they could even use the GUI. The important part is that the image is uploaded **to the Azure Stack Environment** and is **public**